src/proto/plumtree.rs

```diff
@@ -625,6 +646,8 @@ impl<PI: PeerIdentity> State<PI> {
                     sender,
                     Message::Gossip(message.clone()),
                 ));
+            } else {
+                debug!(?id, peer=?sender, "on_graft failed to graft: message not in cache");
```

Comment:
being curious, why is this newly interesting?

Reply:
I found the bug about  nodes dying due to too-large `IHave` messages by backtracking missing messages, and that's one of the lines that helped a lot on the way. So it is not directly related to the change here, but I left it in from the debugging session, because I think it's useful in general as a debug log.